src/evaluators/utility/PHAL_ComputeBasisFunctionsSide_Def.hpp

```diff
@@ -566,7 +431,12 @@ evaluateFields(typename Traits::EvalData workset)
       for (std::size_t iCell=0; iCell < numCells_; ++iCell)
         for (unsigned int icoor=0; icoor<numCellDims; ++icoor)
           for (unsigned int qp=0; qp<numSideQPs; ++qp)
-            normals(sideSet.cellsOnSide(side,iCell),side,qp, icoor) = normals_view(iCell,qp,icoor);
+          {
+            if (useCollapsedSidesets)
+              normals(sideSet.sideSetIdxOnSide(side,iCell),qp, icoor) = normals_view(iCell,qp,icoor);
```

Comment:
I forgot, what was the reason for not making this a kernel?

Reply:
I think there was a reason at one point but the way I've rewritten it, it would now be easy to turn it into a kernel. I'll also add one for populating physPointsCell.

Reply:
okay great thanks